pkg/lib/aws/cloudwatch.go

```diff
@@ -64,3 +82,249 @@ func (c *Client) TagLogGroup(logGroup string, tagMap map[string]string) error {
+	latencyWidgetp50 := createMetricWidget(1, highestY+8, 11, 6, latencyMetric(dashboardName, nameAPI), "median request response time (60s)", "p50", 60, dashboardRegion)
+	// bottom right widget
+	latencyWidgetp99 := createMetricWidget(12, highestY+8, 11, 6, latencyMetric(dashboardName, nameAPI), "p99 of request response time (60s)", "p99", 60, dashboardRegion)
+
+	// append new API metrics widgets to existing widgets
+	currDashboard.Widgets = append(currDashboard.Widgets, apiTitleWidget, statCodeWidget, inFlightWidget, latencyWidgetp50, latencyWidgetp99)
+	currDashboardJSON, err := json.Marshal(currDashboard)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	// upload updated dashboard to cloudwatch
+	_, err = c.CloudWatch().PutDashboard(&cloudwatch.PutDashboardInput{
+		DashboardName: aws.String(dashboardName),
+		DashboardBody: aws.String(string(currDashboardJSON)),
+	})
+	if err != nil {
+		return errors.Wrap(err)
+	}
+	return nil
+}
+
+// CreateDashboard creates a new dashboard (or clears an existing one if it already exists)
+func (c *Client) CreateDashboard(dashboardName string) error {
+
+	//create cloudwatch base body with title
+	cloudwatchBaseBody := CloudWatchDashboard{Start: "-PT1H", PeriodOverride: "inherit", Widgets: []CloudWatchWidget{createTextWidget(7, 0, 10, 1, "# CORTEX MONITORING DASHBOARD")}}
+	cloudwatchBaseBodyJSON, err := json.Marshal(cloudwatchBaseBody)
+	if err != nil {
+		return errors.Wrap(err, "failed to encode cloudwatch base body into json")
+	}
+	_, err = c.CloudWatch().PutDashboard(&cloudwatch.PutDashboardInput{
+		DashboardName: aws.String(dashboardName),
+		DashboardBody: aws.String(string(cloudwatchBaseBodyJSON)),
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to create dashboard: "+dashboardName)
+	}
+
+	return nil
+}
+
+// DeleteDashboard deletes dashboard
+func (c *Client) DeleteDashboard(dashboardName string) error {
+
+	_, err := c.CloudWatch().DeleteDashboards(&cloudwatch.DeleteDashboardsInput{
+		DashboardNames: []*string{aws.String(dashboardName)},
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to delete dashboard: "+dashboardName)
+	}
+	return nil
+}
+
+// DoesDashboardExist will check if dashboard with same name as cluster already exists
+func (c *Client) DoesDashboardExist(dashboardName string) (bool, error) {
+	_, err := c.CloudWatch().GetDashboard(&cloudwatch.GetDashboardInput{
+		DashboardName: aws.String(dashboardName),
+	})
+	if err != nil {
+		if CheckErrCode(err, "ResourceNotFound") {
+			return false, nil
+		}
+		return false, errors.Wrap(err, "dashboard: "+dashboardName)
+	}
+
+	return true, nil
+}
+
+// DeleteAPICloudwatch deletes api and reformats cloudwatch
+func (c *Client) DeleteAPICloudwatch(allAPINames []string, clusterName, apiName string) error {
+
+	//delete old dashboard by creating a new base dashboard
+	err := c.CreateDashboard(clusterName)
+	if err != nil {
+		return errors.Wrap(err, "failed getting API statuses while deleting API")
```

Comment:
this error wrap message is incorrect, feel free to update it or delete it, whichever you think makes most sense